tests/ballerina-test/src/test/java/org/ballerinalang/test/nativeimpl/functions/compression/CompressionTest.java

```diff
@@ -0,0 +1,286 @@
+        String resourceToRead = getAbsoluteFilePath("datafiles/compression/");
+        BString dirPath = new BString(resourceToRead);
+        BString destDir = new BString(null);
+        BValue[] inputArg = {dirPath, destDir};
+        BValue[] returns = BRunUtil.invoke(compileResult, "testZipFile", inputArg);
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertNotNull(returns[0]);
+        Assert.assertEquals(returns[0].stringValue(), "testZipFile");
+    }
+
+    @Test(description = "test unzipping/decompressing a byte array")
+    public void testUnzipBytes() throws IOException, URISyntaxException {
+        String dirPath = getAbsoluteFilePath("datafiles/compression/");
+        byte[] fileContentAsByteArray = Files.readAllBytes(new File(dirPath +
+                File.separator + "test.zip").toPath());
+        BString destDir = new BString(dirPath);
+        BString folderToExtract = new BString("");
+        BBlob contentAsByteArray = new BBlob(fileContentAsByteArray);
+        BValue[] inputArg = {contentAsByteArray, destDir, folderToExtract};
+        BRunUtil.invoke(compileResult, "testUnzipBytes", inputArg);
+
+        File f1 = new File(dirPath + File.separator + "test/");
+        Assert.assertEquals(f1.exists() && f1.isDirectory(), true);
+
+        File f2 = new File(dirPath + File.separator + "test/test.txt");
+        Assert.assertEquals(f2.exists() && !f2.isDirectory(), true);
+
+    }
+
+    @Test(description = "test unzipping/decompressing a byte array without the content as a blob/byte array",
+            expectedExceptions = BLangRuntimeException.class)
+    public void testUnzipBytesWithoutBytes() throws IOException, URISyntaxException {
+        String dirPath = getAbsoluteFilePath("datafiles/compression/");
+        byte[] fileContentAsByteArray = null;
+        BString folderToExtract = new BString("");
+        BString destDir = new BString(dirPath);
+        BBlob contentAsByteArray = new BBlob(fileContentAsByteArray);
+        BValue[] inputArg = {contentAsByteArray, destDir, folderToExtract};
+        BValue[] returns = BRunUtil.invoke(compileResult, "testUnzipBytes", inputArg);
+        Assert.assertNotNull(returns);
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertNotNull(returns[0]);
+        Assert.assertEquals(returns[0].stringValue(), "testUnzipBytes");
+
+    }
+
+    @Test(description = "test unzipping/decompressing a byte array without the destination directory path",
+            expectedExceptions = BLangRuntimeException.class)
+    public void testUnzipBytesWithoutDestDirectory() throws IOException, URISyntaxException {
+        String dirPath = getAbsoluteFilePath("datafiles/compression/");
+        byte[] fileContentAsByteArray = Files.readAllBytes(new File(dirPath +
+                File.separator + "test.zip").toPath());
+        BString destDir = new BString(null);
```

Comment:
I think the correct test would be to assign null to `destDir` (rather than `new BString(null)` ).
It would be better to write this test using ballerina. eg:
```ballerina
function testUnzipFile(string dirPath, string destDir, string folderToExtract) {
    compression:unzipFile(null, null, null);
}
```
Should do the same for all the four newly implemented methods, and for all input paramters separately.